core/src/main/java/org/apache/iceberg/rest/RESTFileScanTaskParser.java

```diff
@@ -87,7 +87,7 @@ public static FileScanTask fromJson(
     if (jsonNode.has(DELETE_FILE_REFERENCES)) {
       List<Integer> indices = JsonUtil.getIntegerList(DELETE_FILE_REFERENCES, jsonNode);
       Preconditions.checkArgument(
-          Collections.max(indices) < allDeleteFiles.size(),
+          indices.isEmpty() || Collections.max(indices) < allDeleteFiles.size(),
```

Comment:
The fix LGTM, i had it sitting on the part 2 pr : https://github.com/apache/iceberg/pull/13400/files#diff-584f9d53626a76efc298a57ada578de6d07c6d0b00f767f85a05e40471426374R90 since Jun 26 :) 

I have one more fix which is in the part 2 pr let me get this out of this as well